### PR TITLE
Make nearest_mult8 implementation concise

### LINKED ISSUE #2: Make nearest_mult8 concise
I'd like to work on this just to dip my toes into the project.
Here is the current implementation of nearest_mult8 function → https://github.com/Feral-Lang/Feral/blob/9a65fe643918eec9969a9e02aee2d6c756b0e770/src/VM/Memory.cpp#L14

I'd like to shorten it to `sz = (sz + 7) & ~7` and build & run tests along the way.

### PR DESCRIPTION
**Pull request by vitkarpov**
Fixes #2

How to test

- MEM_PROFILE=true cmake ..
- feral tests/facto_iterate.fer > facto_iterate.out

I got the following output → https://gist.github.com/vitkarpov/46a3a41d9c511239e38b774e67511083 with both versions (before & after changes applied).

### CODE REVIEW

File: src/VM/Memory.cpp [hunk 1]
```diff
@@ -13,72 +13,7 @@ namespace mem
...
-	else if( sz >= 449 && sz <= 456 ) return 456;
-	else if( sz >= 457 && sz <= 464 ) return 464;
-	else if( sz >= 465 && sz <= 472 ) return 472;
-	else if( sz >= 473 && sz <= 480 ) return 480;
-	else if( sz >= 481 && sz <= 488 ) return 488;
-	else if( sz >= 489 && sz <= 496 ) return 496;
-	else if( sz >= 497 && sz <= 504 ) return 504;
-	else if( sz >= 505 && sz <= 512 ) return 512;
-	return sz;
+	return ( sz > 512 ) ? sz : ( sz + 7 ) & ~7;
```
Comment (vitkarpov) [inline src/VM/Memory.cpp:16]: @Electrux I have a question. Why do we need it as-is when greater than 512? Could you, pls, share some insight?

Comment (Chirag-Khandelwal) [reply src/VM/Memory.cpp:16]: The reason here is that blocks larger than 512 are seldomly needed and such fine management of a (relatively) large block that is not necessary. There is definitely quite some profiling to be done to get a tangible proof regarding that, but from all the programs that I have run yet, none have even surpassed ~256 bytes for a single allocation, much less 512. So managing anything above that kind of seams like a waste.
Also, I used this (Cpython) as a reference for that decision: https://github.com/python/cpython/blob/08050e959e6c40839cd2c9e5f6a4fd1513e3d605/Objects/obmalloc.c#L787

Comment (Chirag-Khandelwal) [reply src/VM/Memory.cpp:16]: If we keep manage large sized blocks, whose allocation/deallocation is far rare, the whole purpose of custom allocator becomes moot - barely any performance gains.
Plus, if there are a large number of allocations > 512 bytes, new pools would be allocated much more frequently anyway.
At the end of the day, it is all about a particular application and its memory usage for the most part, but from what I have seen, this works just fine (at least for now). 😄

**Comment by Chirag-Khandelwal** [comment]
Thanks a lot 😁
It was a really cool optimization!

File: src/VM/Memory.cpp (hunk 1 above)
Comment (vitkarpov) [reply src/VM/Memory.cpp:16]: Great, thanks for the explanation and python reference. I'm really into learning that kind of stuff.

**Comment by vitkarpov** [comment]
My pleasure, looking forward to a next problem. I'd like to get to know more about the system. If you have something in mind, please, ping me.

**Pull request merged**